packages/packages.go

```diff
@@ -95,6 +102,46 @@ type QFEPackage struct {
 	Caption, Description, HotFixID, InstalledOn string
 }
 
+// CommandRunner will execute the commands and return the results of that
+// execution.
+type CommandRunner interface {
+
+	// RunCommand takes precreated exec.Cmd and returns the results of execution.
+	RunCommand(command *exec.Cmd) ([]byte, error)
+
+	// Run takes string arguments of command to be executed
+	// and returns the results of execution.
+	Run(command string, args ...string) ([]byte, error)
```

Comment:
It doesn't look like this function is ever called. Would it be possible to use this function instead of RunCommand? It seems redundant for callers to write `Runner.RunCommand(exec.Command(...))` instead of `Runner.RunCommand(...)`. Is it possible, to remove `RunCommand`, change `RunWithPty` to take `command string, args ...string` and have the CommandRunner wrap `exec.Command` too?

Reply:
i added ```Run(...)``` to provide a simpler variation of ```RunCommand()```, that i was using in tests(for test setup), but ultimately didn't end up using it, but i left it with the idea that it can be used if a client of this CommandRunner does not want to create exec.Cmd on their own.
if i were to remove, i would prefer removing ```Run(...)``` and keep the signatures of ```RunWithPty() && RunCommand()``` to accept ```exec.Cmd```, since that is more intuitive(clients will read this as ```create exec.Cmd with exec.Command(...) and send it to commandRunner to execute the Cmd```)

Reply:
this is a specific example of using exec.Command in tests.
https://github.com/adjackura/osconfig/blob/5354872db63e6a6e86ba6c0aacb3253fabab95f0/inventory/packages/yum_test.go#L77

added here for reference